features/activity/cancel_try_cancel/feature.workflow.ts

```diff
@@ -0,0 +1,54 @@
+import type { feature } from './feature';
+import { ActivityFailure, ApplicationFailure } from '@temporalio/common';
+import {
+  proxyActivities,
+  sleep,
+  CancellationScope,
+  CancelledFailure,
+  ActivityCancellationType,
+  defineSignal,
+  setHandler,
+  condition,
+} from '@temporalio/workflow';
+
+// Allow 4 retries with no backoff
+const { cancellableActivity } = proxyActivities<typeof feature.activities>({
+  startToCloseTimeout: '1 minute',
+  heartbeatTimeout: '5 seconds',
+  // Disable retry
+  retry: { maximumAttempts: 1 },
+  cancellationType: ActivityCancellationType.TRY_CANCEL,
+});
+
+export const activityResultSignal = defineSignal<[string]>('activity-result');
+
+export async function workflow(): Promise<void> {
+  try {
+    await CancellationScope.cancellable(async () => {
+      // Start activity
+      const actPromise = cancellableActivity();
+
+      // Sleep for smallest amount of time (force task turnover)
+      await sleep(1);
+
+      // Cancel activity and await it
+      CancellationScope.current().cancel();
+      await actPromise;
+    });
+    throw new Error('No error');
+  } catch (e) {
+    // Confirm the activity was cancelled
+    if (!(e instanceof ActivityFailure) || !(e.cause instanceof CancelledFailure)) {
+      const error = e instanceof Error ? e : new Error(`${e}`);
+      throw new ApplicationFailure('Expected cancellable', null, true, undefined, error);
+    }
+  }
+
+  // Confirm signal is received saying the activity got the cancel
+  let activityResult: string | undefined;
+  setHandler(activityResultSignal, (v) => void (activityResult = v));
+  await condition(() => !!activityResult, 10000);
```

Comment:
Do we have to set a timeout here?
Wouldn't it be better to implement the test in a naive manner and set the timeout in the runner as a general strategy?

Reply:
+1 to that. Would be nice to have default timeout in runner than can be overriden, and just terminates the workflow with a "timed out" reason or something.

Reply:
nit:
```ts
const activityResult = await new Promise<string>((resolve) => setHandler(activityResultSignal, resolve));
```

Or with timeout:
```ts
const activityResult = await new Promise<string>((resolve, reject) => {
  setHandler(activityResultSignal, resolve);
  sleep(10000).then(() => reject(ApplicationFailure.nonRetryable('timed out')));
});
```

Reply:
> Do we have to set a timeout here?

No, I just figure it's best practice for anything you expect to happen soon

>  Wouldn't it be better to implement the test in a naive manner and set the timeout in the runner as a general strategy?

Yes. If we don't already I can make that happen.

> nit:

Ok

Reply:
Removed timeouts, put 1 min default workflow execution timeouts on all harnesses, changed activity result